src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -407,12 +409,28 @@ public CodeFixProvider GetSuppressionFixer(string language, IEnumerable<string>
             foreach (var fixer in allFixers)
             {
                 await extensionManager.PerformActionAsync(fixer, () => fixer.RegisterCodeFixesAsync(context) ?? SpecializedTasks.EmptyTask).ConfigureAwait(false);
-                if (!fixes.Any())
+                foreach (var fix in fixes)
                 {
-                    continue;
-                }
+                    if (!fix.Action.PerformFinalApplicabilityCheck)
+                    {
+                        return true;
+                    }
 
-                return true;
+                    // Have to see if this fix is still applicable.  Jump to the foreground thread
+                    // to make that check.
```

Comment:
Is this the typical way we jump to the foreground? I thought most of VS use JoinableTaskFactory to avoid deadlocks?